src/stdune/temp.ml

```diff
@@ -1,16 +1,69 @@
+let try_times n ~f =
+  assert (n > 0);
+  let rec loop n =
+    if n = 1 then
+      f n
+    else
+      match f n with
+      | exception _ -> loop (n - 1)
+      | r -> r
+  in
+  loop n
+
+let prng = lazy (Random.State.make_self_init ())
+
+let temp_file_name ?temp_dir ~prefix ~suffix () =
+  let temp_dir =
+    match temp_dir with
+    | None -> Filename.get_temp_dir_name ()
+    | Some t -> t
+  in
+  let rnd = Random.State.bits (Lazy.force prng) land 0xFFFFFF in
+  Filename.concat temp_dir (Printf.sprintf "%s%06x%s" prefix rnd suffix)
+
 let tmp_files = ref Path.Set.empty
 
+let tmp_dirs = ref Path.Set.empty
+
+external open_desc : string -> open_flag list -> int -> int = "caml_sys_open"
```

Comment:
We can't use these functions as they are not part of the public stdlib API. We should use the `Unix` functions instead

Reply:
Replace with `Unix`.